### PR TITLE
Refactor: move template and runner classes into their own files

### PR DESCRIPTION
**Pull request by miparnisari**
*Issue #, if available:* N/A

*Description of changes:* Refactor: move template and runner classes into their own files :recycle:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/__init__.py [hunk 1]
```diff
@@ -30,8 +22,6 @@
 
 LOGGER = logging.getLogger(__name__)
 
-# pylint: disable=too-many-lines
```
Comment (kddejong) [inline src/cfnlint/__init__.py:33]: 👍

**Comment by kddejong** [comment]
Something like
```
@refactored('Template is refactored and deprecated. Please use Template in cfnlint.template')
class Template(_Template):
    """ Refactored class Template """
```

**Comment by miparnisari** [comment]
Ok so, doing this

```python
import cfnlint.core

# The path to the file to check
filename = '/Users/mparnisa/workspace-sync/cfn-python-lint/test/fixtures/templates/good/conditions/and.yaml'

# Load the YAML file
template = cfnlint.decode.cfn_yaml.load(filename)

temp = cfnlint.core.Template(filename, template)
temp.build_graph()

temp2 = cfnlint.template.Template(filename, template)
temp2.build_graph()

```

prints 

```
/usr/local/bin/python2.7 /Users/mparnisa/workspace-sync/cfn-python-lint/test/integration/test_customer_integration.py -t test/fixtures/templates/bad/resources_circular_dependency_2.yaml -g
/Users/mparnisa/workspace-sync/cfn-python-lint/src/cfnlint/decorators/refactored.py:16: FutureWarning: Template is refactored and deprecated. Please use Template in cfnlint.template
  warnings.warn(message, FutureWarning)

Process finished with exit code 0

```

I don't understand why I didn't need to add `import cfnlint.template`.

File: src/cfnlint/core.py [hunk 2]
```diff
@@ -5,20 +5,30 @@
...
 import cfnlint.decode
 import cfnlint.maintenance
 from cfnlint.helpers import REGIONS
-
+from cfnlint.runner import Runner as _Runner
 
 LOGGER = logging.getLogger('cfnlint')
 DEFAULT_RULESDIR = os.path.join(os.path.dirname(__file__), 'rules')
 
+@refactored('Template is refactored and deprecated. Please use Template in cfnlint.template')
```
Comment (kddejong) [inline src/cfnlint/core.py:25]: Previously it was `from cfnlint import Template`. I think these should be moved to the `src/cfnlint/__init__.py` file instead.

File: src/cfnlint/core.py [hunk 3]
```diff
@@ -5,20 +5,30 @@
...
+class Template(_Template):
+    """ Refactored class Template """
+
+@refactored('Runner is refactored and deprecated. Please use Runner in cfnlint.runner')
+class Runner(_Runner):
```
Comment (kddejong) [inline src/cfnlint/core.py:30]: Same it was `from cfnlint import Runner`. I think these should be moved to the `src/cfnlint/__init__.py` file instead.

**Pull request merged**